doc: add important note of the introduction of upgrade path crd validation (#1081)
longhorn/website: content/docs/1.6.5/important-notes/_index.md

---
title: Important Notes
weight: 1
---

This page lists important notes for Longhorn v{{< current-version >}}.
Please see [here](https://github.com/longhorn/longhorn/releases/tag/v{{< current-version >}}) for the full release note.

- [General](#general)

>   - [CRD Upgrade Validation](#crd-upgrade-validations)

Reviewer: ```suggestion
  - [CRD Upgrade Validation](#crd-upgrade-validation)
```

I think, it is breaking the CI.

Author: Thanks!
Fixed, please take a look again.

## General

### CRD Upgrade Validation

> During the upgrade process, the Custom Resource Definition (CRD) may be applied after the new Longhorn manager has started. This ensures that the controller does not process objects with deprecated data or fields. As a result, the Longhorn manager might fail at the start of the upgrade.

Reviewer: ```suggestion
During the upgrade process, the Custom Resource Definition (CRD) may be applied after the new Longhorn manager has started. This sequencing ensures that the controller does not process objects with deprecated data or fields. However, this can result in the Longhorn manager failing during the initial upgrade phase if the CRD has not been applied yet.
```

Author: fixed.

> During the upgrade process, the Custom Resource Definition (CRD) may be applied after the new Longhorn manager starts. This ensures that the controller does not process objects containing deprecated data or fields. As a result, the Longhorn manager may fail at the beginning of the upgrade.

Reviewer: ```suggestion
During the upgrade process, the Custom Resource Definition (CRD) may be applied after the new Longhorn manager has started. This ensures that the controller does not process objects with deprecated data or fields. As a result, the Longhorn manager might fail at the start of the upgrade.
```

Author: fixed.

> If the Longhorn manager crashes during the upgrade, check the logs to confirm whether the failure is due to the CRD not being applied yet. The logs may contain error messages similar to the following example:

Reviewer: ```suggestion
If the Longhorn manager crashes during the upgrade, check the logs to determine if the failure is due to the CRD not being applied. In such cases, the logs may contain error messages similar to the following:
```

Author: fixed.

> If the Longhorn manager crashes during the upgrade, check the logs to determine whether the failure is due to the CRD not being applied yet. The logs may contain error messages similar to the following example:

Reviewer: ```suggestion
If the Longhorn manager crashes during the upgrade, check the logs to confirm whether the failure is due to the CRD not being applied yet. The logs may contain error messages similar to the following example:
```
